src/agent/src/main.rs

```diff
@@ -412,11 +415,19 @@ async fn start_sandbox(
             );
         } else {
             init_attestation_components(logger, config)?;
+            gc_enabled = Some(gc_procs);
         }
     }
 
     // vsock:///dev/vsock, port
-    let mut server = rpc::start(sandbox.clone(), config.server_addr.as_str(), init_mode).await?;
+    let mut server = rpc::start(
+        sandbox.clone(),
+        config.server_addr.as_str(),
+        init_mode,
+        gc_enabled,
```

Comment:
Maybe we could directly pass `Some(CDHClient)` as parameter here? which would help to avoid splited condition judgement on different places.

Reply:
Good idea. I moved the initialization of the CDHClient to `init_attestation_components` so we can create the CDHClient right after starting the CDIH.